### ISSUE TITLE
Refactor SignInPage providers based on BackstageIdentityApi

### ISSUE THREAD

**Issue by Rugvip** [spotify/backstage#1776]
## Feature Suggestion

All of the BackstageIdentity-based sign-in providers are very similar, let's refactor them to share implementation.

For example https://github.com/spotify/backstage/blob/master/packages/core/src/layout/SignInPage/googleProvider.tsx, https://github.com/spotify/backstage/blob/master/packages/core/src/layout/SignInPage/gitlabProvider.tsx.

## Possible Implementation

Refactor to have a common provider implementation for any Utility API that implements the BackstageIdentityApi, such as `googleAuthApiRef` and `gitlabApiRef`. The varying factors between the providers would be the `id`, `title`, `message`, and `apiRef`. As a part of the refactor we'd also like to add the possibility for providers to be passed from the outside, something like an `identityProviders` prop that is a list of providers with the varying fields provided, e.g.

```tsx
const customProviders = [{
  id: 'my-provider',
  title: 'My Provider',
  message: 'Sign in with My Provider',
  apiRef: myAuthApiRef,
}]

const page = <SignInPage identityProviders={customProviders} />
```

## Context

https://discordapp.com/channels/687207715902193673/687235481154617364/738053909884698705

**Comment by kingasieminiak** [comment]
Hi @Rugvip! I was following conversations on the support channel, I can take the issue.

**Comment by Rugvip** [comment]
@kingasieminiak Awesome, thanks! 😁 🎉

**Comment by kingasieminiak** [comment]
@Rugvip I see that SignInPage and UserSettings components share the same source of providers (configApi). This is great but it still gives a closed list of providers e.g. 

```
{providers.includes('oauth2') && (
    <OIDCProviderSettings
        title="YourOrg"
        apiRef={oauth2ApiRef}
        icon={Star}
    />
)}
```

I think would be great for those components to share the same, more flexible, configuration. Although, I see the providers don't share the same "settings components": (OIDCProviderSettings, OAuthProviderSettings)  so they could be passed from the outside (as a part of config), but I am wondering if this is overcomplicated.

**Comment by Rugvip** [comment]
@kingasieminiak Tbh the providers on the sign-in-page shouldn't be collected from config in the long run, since sign-in and the auth delegation are separate concerns. The set of 3rd party access providers you want to support is likely not going to be the same as the ones you want to use for sign-in.

The sign-in providers should also not really be concerned with OAuth/OIDC at all, will post a comment on the PR about that :grin:

**Comment by stefanalund** [comment]
Can this be closed? Looks like #1816 took care of it.

**Comment by Rugvip** [comment]
Yep!
